purchases/src/main/kotlin/com/revenuecat/purchases/galaxy/conversions/StoreProductConversions.kt

```diff
@@ -25,20 +40,108 @@ internal fun ProductVo.toStoreProduct(): StoreProduct {
         null
     }
 
+    val subscriptionOptions: SubscriptionOptions?
+    val defaultOption: SubscriptionOption?
+    if (productType == ProductType.SUBS && standardBillingPeriod != null) {
+        val subscriptionOption = GalaxySubscriptionOption(
+            id = this.itemId,
+            pricingPhases = this.createPricingPhases(
+                promotionEligibilities = promotionEligibilities,
+                standardBillingPeriod = standardBillingPeriod,
+                standardPrice = standardPrice,
+            ),
+            tags = emptyList(), // Tags are unsupported on the Galaxy Store
+            presentedOfferingContext = null,
+            purchasingData = GalaxyPurchasingData.Product(productId = productId, productType = productType),
+            installmentsInfo = null,
+        )
+
+        subscriptionOptions = SubscriptionOptions(subscriptionOptions = listOf(subscriptionOption))
+        defaultOption = subscriptionOption
+    } else {
+        subscriptionOptions = null
+        defaultOption = null
+    }
+
     return GalaxyStoreProduct(
-        id = this.itemId,
-        type = type,
-        price = this.createPrice(),
+        id = productId,
+        type = productType,
+        price = standardPrice,
         name = this.itemName,
         title = this.itemName,
         description = this.itemDesc,
-        period = period,
-        subscriptionOptions = null,
-        defaultOption = null,
+        period = standardBillingPeriod,
+        subscriptionOptions = subscriptionOptions,
+        defaultOption = defaultOption,
         presentedOfferingContext = null,
     )
 }
 
+private fun ProductVo.createPricingPhases(
+    promotionEligibilities: List<PromotionEligibilityVo>?,
+    standardBillingPeriod: Period,
+    standardPrice: Price,
+): List<PricingPhase> {
+    val pricingPhases: MutableList<PricingPhase> = mutableListOf()
+    val type = this.type.createRevenueCatProductTypeFromSamsungIAPTypeString()
+
+    val eligibilityPricings: MutableSet<String> = HashSet()
+    promotionEligibilities
+        ?.filter { it.itemId == this.itemId }
+        ?.mapTo(eligibilityPricings) { it.pricing }
+
+    if (
```

Comment:
We still haven't seen this case appear in our testing, but since it's possible through the API, it's probably worth handling in the code :)